code/modules/cargo/packs/_packs.dm

```diff
@@ -122,3 +122,15 @@
 	name = "[purchaser]'s Materials Order"
 	src.cost = cost
 	src.contains = contains
+
+/datum/supply_pack/custom/minerals/fill(obj/structure/closet/crate/C)
+	. = ..()
+	//Remove our material sheets from SSstock_market's materials_quantity equal to the quantity within the crate.
+	for(var/obj/item/stack/sheet/possible_stack as anything in contains)
+		if(!ispath(possible_stack, /obj/item/stack/sheet))
+			continue
+		if(possible_stack.material_type)
+			if(!SSstock_market.materials_quantity[possible_stack.material_type])
+				continue
+			SSstock_market.materials_quantity[possible_stack.material_type] -= possible_stack.amount
+			SSstock_market.materials_prices[possible_stack.material_type] += round((SSstock_market.materials_prices[possible_stack.material_type]) * (possible_stack.amount / (SSstock_market.materials_quantity[possible_stack.material_type] - possible_stack.amount)))
```

Comment:
```suggestion
	var/material_type
	var/available_quantity
	var/obj/item/stack/sheet/ordered_stack

	for(var/obj/item/stack/sheet/possible_stack as anything in contains)
		material_type = initial(possible_stack.material_type)

		//in case we ordered more than what's in the market at the time due to market fluctuations
		//we find the min of what was ordered & what's actually available in the market at this point of time
		available_quantity = min(contains[possible_stack], SSstock_market.materials_quantity[material_type])
		if(!available_quantity)
			continue

		//spawn the ordered stack inside the crate
		ordered_stack = new possible_stack(C, available_quantity)
		if(admin_spawned)
			ordered_stack.flags_1 |= ADMIN_SPAWNED_1

		//Prices go up as material quantity becomes scarce
		SSstock_market.materials_prices[material_type] += round((SSstock_market.materials_prices[material_type]) * (available_quantity / (SSstock_market.materials_quantity[material_type] - available_quantity)))
		//We decrease the quantity only after adjusting our prices for accurate values
		SSstock_market.materials_quantity[material_type] -= available_quantity
```

Reply:
In applying this suggestion, it spawns an empty crate. I didn't see the runtime you encountered, but I did find an alternative fix for the time being.

Reply:
Empty crates? I tested this code locally just now and it works fine, i even debugged the code in the video and all stacks are accounted for

https://github.com/tgstation/tgstation/assets/110812394/a4dfaff8-c452-46d0-883b-80da98943485

This suggestion applies 2 points which you have missed
- It checks the ordered quantity against the quantity available in the market and finds the min so we don't end up  putting more stacks in the create than what's actually available in the market
- It checks for divide by zero error when changing the cost

Are you sure you saw a empty crate?

Reply:
I have no opinion on this but brother why are you defining the vars outside the loop this is not atmos code

Reply:
Yup, 100%, on my local with that code suggestion I received an empty crate after buying ~475 bluespace crystals from the market. I intentionally bought around the market cap to notice a visible difference between the market quantity and not. 

There a chance what happened was a fluke on my part? Still it didn't provide a runtime so I didn't think to screenshot it earlier before I just recompiled.

Reply:
> I have no opinion on this but brother why are you defining the vars empty outside the loop this is not atmos code

It's because a for loop will redefine the var over and over again every iteration? But yeah i can move it inside the loop no worries

Reply:
> up, 100%, on my local with that code suggestion I received an empty crate after buying ~475 bluespace crystals

OHHH. Ok i understand your problem now. Yup that is a problem on my end i can fix it just give me 10 minutes. Yup you get an empty crate because the maximum possible stack size is 50 so any value abbove that will cause problems.

Thank you i can fix it just give me a sec

Reply:
Never mind it got merged. This discussion can be resolved